File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Method - Override 01.fs
module Module

type T() =
    inherit InheritedClass()

    override this.MPublic() = base.MPublic()
    override this.MProtected() = base.MProtected()
    override this.MPrivate() = base.MPrivate()

    override this.NonVirtual() = base.NonVirtual()
    override this.NonVirtualOverriden() = base.NonVirtualOverriden()

File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Method - Override 01.cs
public abstract class BaseClass
{
    public virtual void MPublic() { }
    protected virtual void MProtected() { }
    private virtual void MPrivate() { }

    public void NonVirtual() { }
    public void NonVirtualOverriden() { }
}

public class InheritedClass : BaseClass
{
    public override void MPublic() { }
    protected override void MProtected() { }
    private override void MPrivate() { }

    public override void NonVirtualOverriden() { }
}
